.NET/Microsoft.Recognizers.Text.DateTime/Chinese/Parsers/DateParser.cs

```diff
@@ -189,22 +189,39 @@ protected DateTimeResolutionResult ParseImplicitDate(string text, DateObject ref
                     futureDate = DateObject.MinValue.SafeCreateFromValue(year, month + 1, day);
                     pastDate = DateObject.MinValue.SafeCreateFromValue(year, month - 1, day);
                 }
+
                 else
                 {
-                    futureDate = DateObject.MinValue.SafeCreateFromValue(year, month, day);
-                    pastDate = DateObject.MinValue.SafeCreateFromValue(year, month, day);
+                    if (!IsLeapYear(year) && IsFeb29th(month, day))

```

Comment:
Could you please add some comments containing some examples, the logic is not quite easy to understand.